web/src/beta/features/Editor/Visualizer/hooks.ts

```diff
@@ -87,8 +91,14 @@ export default ({
     [selected],
   );
 
-  const layers = useMemo(() => processLayers(nlsLayers), [nlsLayers]);
-
+  const layers = useMemo(
+    () =>
+      processLayers(nlsLayers)?.map(layer => ({
+        ...layer,
+        hidden: false,
+      })),
+    [nlsLayers],
+  );
```

Comment:
On all tabs BUT story tab and publish/story we will want hidden to be how it is set in the Map tab. As far as I can tell this will force all layers to be visible from the start, which isn't good.

Then below, you have layersData state + a useEffect processing layers more, which I find these two being redundant and adding extra code.

I think better is to just do it all here in this layers useMemo.
Something like this (code is probaly not completely correct):
```suggestion
  const layers = useMemo(
    () => {
    let processedLayers = processLayers(nlsLayers)
    if (!!showStoryPanel) { 
       return processedLayers;
   };
   return processedLayers?.map(layer => ({
        ...layer,
        hidden: !currentPage?.layersIds?.includes(layer.id),
      })
     },
    [nlsLayers, currentPage?.layersIds],
  );
```